### ISSUE TITLE
`MissingPrimaryKey` false positive on resources using `AshEventLog.EventLog`

### ISSUE THREAD

**Issue by jeroen11dijk** [leonqadirie/ash_credo#132]
**Check:** `AshCredo.Check.Warning.MissingPrimaryKey`
**ash_credo version:** 0.12.1

## Summary

`MissingPrimaryKey` flags resources that use the `AshEventLog.EventLog` extension, even though the extension *adds* a primary key via a Spark transformer at compile time. The resource is fully functional — the check is statically analyzing the source file and cannot see the transformer's contribution.

## Reproduction

```elixir
defmodule MyApp.AuditTrail.Event do
  @moduledoc false
  use Ash.Resource,
    domain: MyApp.AuditTrail,
    data_layer: AshPostgres.DataLayer,
    extensions: [AshEventLog.EventLog]

  postgres do
    table "events"
    repo MyApp.Repo
  end

  event_log do
    primary_key_type(:uuid_v7)
    record_id_type(:uuid)
    persist_actor_primary_key(:user_id, MyApp.Accounts.User)
    public_fields(:all)
  end
end
```

Running:

```
mix credo --strict --checks AshCredo.Check.Warning.MissingPrimaryKey
```

Output:

```
[W] ↗ Resource is missing a primary key.
      lib/my_app/audit_trail/resources/event.ex:3 #(MyApp.AuditTrail.Event)
```

`Ash.Resource.Info.primary_key(MyApp.AuditTrail.Event)` returns `[:id]` at runtime — the PK is generated by `AshEventLog.EventLog.Transformers.SetupEventLog` (see `deps/ash_event_log/lib/event_log/transformers/setup_event_log.ex:13` `add_primary_key/3`).

## Root cause

`AshCredo.Check.Warning.MissingPrimaryKey.check_for_primary_key/2` walks the AST of the source file and looks for one of:

- `uuid_primary_key`, `uuid_v7_primary_key`, `integer_primary_key` entities in `attributes`
- An `attribute` entity with `primary_key?: true`
- A `belongs_to` with `primary_key?: true`

The `event_log do ... end` block is not inspected. There is no way for AST-only analysis to know an extension contributes a primary key.

## Suggested fixes (any one would resolve)

1. **Recognize known PK-providing extensions.** Maintain a list (`AshEventLog.EventLog`, possibly others) and skip the check when the resource declares `extensions: [...]` containing one of them.
2. **Use a compiled-project check.** Other AshCredo checks (e.g. `UnknownAction`, `MissingCodeInterface`) already require compilation and use `Ash.Resource.Info.primary_key/1`. The same approach here would eliminate false positives for any extension-supplied PK.
3. **Provide a check-level escape hatch.** Today the only workaround is per-file exclusion via the credo `files: %{excluded: [...]}` option, which is opaque (a future reader can't tell *why* a file is excluded without grepping). A `excluded_extensions: [AshEventLog.EventLog]` option would be more declarative.

Option 2 is the most robust — it fixes this class of false positive in general, not just for `AshEventLog`.
